core/src/main/java/org/infinispan/distribution/impl/L1ManagerImpl.java

```diff
@@ -140,7 +138,7 @@ public void addRequestor(Object key, Address origin) {
    }
 
    @Override
-   public Future<Object> flushCache(Collection<Object> keys, Address origin, boolean assumeOriginKeptEntryInL1) {
+   public Future<?> flushCache(Collection<Object> keys, Address origin, boolean assumeOriginKeptEntryInL1) {
```

Comment:
It's a missed opportunity to deal with a `CompletableFuture` internally and externally only provide Future :neutral_face: 

AFAIK, L1Manager is an internal component and you should be relatively free to change return type to `CompletableFuture` and hence enable the caller to do anything other than blocking on the future's get(), e.g. map, compose...etc.